sql/server.go

```diff
@@ -159,11 +159,15 @@ func (s server) execStmt(stmt parser.Statement, params parameters, planMaker *pl
 				case parser.DString:
 					row.Values = append(row.Values, driver.Datum{StringVal: (*string)(&vt)})
 				case parser.DDate:
-					s := vt.String()
-					row.Values = append(row.Values, driver.Datum{StringVal: &s})
+					row.Values = append(row.Values, driver.Datum{TimeVal: &driver.Datum_Timestamp{
+						Sec:  vt.Unix(),
+						Nsec: uint32(vt.Nanosecond()),
+					}})
 				case parser.DTimestamp:
-					s := vt.String()
-					row.Values = append(row.Values, driver.Datum{StringVal: &s})
+					row.Values = append(row.Values, driver.Datum{TimeVal: &driver.Datum_Timestamp{
+						Sec:  vt.Unix(),
+						Nsec: uint32(vt.Nanosecond()),
+					}})
 				case parser.DInterval:
 					s := vt.String()
 					row.Values = append(row.Values, driver.Datum{StringVal: &s})
```

Comment:
scratch that; we should do this, but it's not that trivial to implement. if we return it as an int, we have no way of knowing it was a DInterval. We'll need a special wire type, at least.

Gonna leave this for a subsequent change.